fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/synthetic/SyntheticRecordPlannerTest.java

```diff
@@ -741,14 +741,67 @@ public void aggregateJoinIndex() throws Exception {
         }
     }
 
+    @Test
+    public void concatenatedFields() throws Exception {
```

Comment:
I don't feel strongly, but suspect this could be a parameterized version of `multiFieldKeys`.

All these tests need plausibility assertions on `JoinedRecordPlan.queries`. Specifically, that they don't do full table scans. I believe this test will fail that.